analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -147,4 +147,23 @@ public override void ExecutionCompleted()
             ReportIssue(operation, operation.Syntax.ToString());
         }
     }
+
+    private static ProgramState SetConstraintForObjectCreation(ProgramState state, IObjectCreationOperationWrapper objectCreation) =>
+        FirstArgumentAsCollection(objectCreation) switch
+        {
+            { } firstArgument when ConstraintFromArgument(state, firstArgument) is { } constraint => state.SetOperationConstraint(objectCreation, constraint),
+            { } => state,
+            _ => state.SetOperationConstraint(objectCreation, CollectionConstraint.Empty)
+        };
+
+    private static IArgumentOperationWrapper? FirstArgumentAsCollection(IObjectCreationOperationWrapper objectCreation) =>
+        objectCreation.Arguments.Select(x => x.AsArgument()).SingleOrDefault(x => x?.Parameter.Ordinal == 0) is { } firstArgument
```

Comment:
It's not the first one when calling stuff out of order. It's argument of first parameter, but the idea should be that we want the collection argument
```suggestion
        objectCreation.Arguments.Select(x => x.AsArgument()).SingleOrDefault(x => x?.Parameter.Ordinal == 0) is { } collectionArgument
```